src/analysis/special_functions/stirling.lean

```diff
@@ -0,0 +1,238 @@
+begin
+  simp only [stirling_seq, factorial_one, cast_one, mul_one, one_div, pow_one, mul_inv_rev,
+    inv_inv],
+end
+
+/-- Define `log_stirling_seq n` as the log of `stirling_seq n`. -/
+noncomputable def log_stirling_seq (n : ℕ) : ℝ := log (stirling_seq n)
+
+/--
+We have the expression
+`log_stirling_seq (n + 1) = log(n + 1)! - 1 / 2 * log(2 * n) - n * log ((n + 1) / e)`.
+-/
+lemma log_stirling_seq_formula (n : ℕ) : log_stirling_seq n.succ =
+  log n.succ.factorial - 1 / 2 * log (2 * n.succ) - n.succ * log (n.succ / exp 1) :=
+begin
+  have h3, from sqrt_ne_zero'.mpr (mul_pos two_pos $ cast_pos.mpr (succ_pos n)),
+  have h4 : 0 ≠ ((n.succ : ℝ) / exp 1) ^ n.succ, from
+    ne_of_lt (pow_pos (div_pos (cast_pos.mpr n.succ_pos ) (exp_pos 1)) n.succ),
+  rw [log_stirling_seq, stirling_seq, log_div, log_mul, sqrt_eq_rpow, log_rpow, log_pow],
+  { linarith },
+  { refine (zero_lt_mul_left two_pos).mpr _,
+    rw ←cast_zero,
+    exact cast_lt.mpr (succ_pos n), },
+  { exact h3, },
+  { exact h4.symm, },
+  { exact cast_ne_zero.mpr n.succ.factorial_ne_zero, },
+  { apply (mul_ne_zero h3 h4.symm), },
+end
+
+/--
+The sequence `log_stirling_seq (m + 1) - log_stirling_seq (m + 2)` has the series expansion
+   `∑ 1 / (2 * (k + 1) + 1) * (1 / 2 * (m + 1) + 1)^(2 * (k + 1))`
+-/
+lemma log_stirling_seq_diff_has_sum (m : ℕ) :
+  has_sum (λ k : ℕ, (1 : ℝ) / (2 * k.succ + 1) * ((1 / (2 * m.succ + 1)) ^ 2) ^ k.succ)
+  (log_stirling_seq m.succ - log_stirling_seq m.succ.succ) :=
+begin
+  change
+    has_sum ((λ b : ℕ, 1 / (2 * (b : ℝ) + 1) * ((1 / (2 * m.succ + 1)) ^ 2) ^ b) ∘ succ) _,
+  rw has_sum_nat_add_iff 1,
+  convert (has_sum_log_one_add_inv $ cast_pos.mpr (succ_pos m)).mul_left ((m.succ : ℝ) + 1 / 2),
+  { ext k,
+    rw [← pow_mul, pow_add],
+    push_cast,
+    have : 2 * (k : ℝ) + 1 ≠ 0, {norm_cast, exact succ_ne_zero (2*k)},
+    have : 2 * ((m : ℝ) + 1) + 1 ≠ 0, {norm_cast, exact succ_ne_zero (2*m.succ)},
+    field_simp,
+    ring },
+  { have h : ∀ (x : ℝ) (hx : x ≠ 0), 1 + x⁻¹ = (x + 1) / x,
+    { intros, rw [_root_.add_div, div_self hx, inv_eq_one_div], },
+    simp only [log_stirling_seq_formula, log_div, log_mul, log_exp, factorial_succ, cast_mul,
+      cast_succ, cast_zero, range_one, sum_singleton, h] { discharger :=
+      `[norm_cast, apply_rules [mul_ne_zero, succ_ne_zero, factorial_ne_zero, exp_ne_zero]] },
+    ring },
+  { apply_instance }
+end
+
+/-- The sequence `log_stirling_seq ∘ succ` is monotone decreasing -/
+lemma log_stirling_seq'_antitone : antitone (log_stirling_seq ∘ succ) :=
+begin
+  apply antitone_nat_of_succ_le,
+  intro n,
+  rw [← sub_nonneg, ← succ_eq_add_one],
+  refine (log_stirling_seq_diff_has_sum n).nonneg _,
+  norm_num,
+  simp only [one_div],
+  intro m,
+  refine mul_nonneg _ _,
+  all_goals {refine inv_nonneg.mpr _, norm_cast, exact (zero_le _)},
+end
+
+/--
+We have the bound  `log_stirling_seq n - log_stirling_seq (n+1) ≤ 1/(2n+1)^2* 1/(1-(1/2n+1)^2)`.
```

Comment:
I find this comment not to be very enlightening and it does not match the statement (there is one set of parentheses missing).

Reply:
Change the comment.